src/authoring/hooks/use-authoring-preview.ts

```diff
@@ -0,0 +1,39 @@
+import { useRef } from "react";
+
+export interface AuthoringPreview {
+  openPreview: (branch: string, unit: string) => void;
+  reloadAllPreviews: () => void;
+}
+
+export const useAuthoringPreview = (): AuthoringPreview => {
+  const windowsRef = useRef<Window[]>([]);
```

Comment:
An improvement perhaps could be to make this work with a context, so the list of windows would be more global. That way the single authoringPreview object wouldn't have to be passed down through the component tree.

Reply:
The new PR on your list is to switch everything to contexts. :)